 # Some Simple models

Import modules

In [1]:
import pandas as pd
import numpy as np

Read data

In [2]:
df = pd.read_csv('consumption.csv')

In [13]:
print(df)

                                        meter_id  2017-01-01 00:00:00  \
0     0xa62b9f23553ff183f61e2bf943aab3d5983d02d7                  NaN   
1     0x459c834d1f6cfb5b734b82aa9f5410fa97fb70da                  NaN   
2     0x4a1ed36825360a058cec2bdd409fc2459e1ce54f                  NaN   
3     0x5b76d3c0e0aefc6e0a8d1d031f96388a23263407                  NaN   
4     0x943ebe39ef2be6ef807c42c5a647e27112ca5b0f                  NaN   
...                                          ...                  ...   
3243  0x7dd7a7b8ee1bec7c44b24f738c752482f6161065                0.011   
3244  0xfdaf9f857621ec06f2cf801f42a020a322835090                0.015   
3245  0xd28f2f001e0cd4d6c121a3cb2e1427207e170e18                0.222   
3246  0x47218b46abb2fcaade487a211911406dc6e13730                0.195   
3247  0xcd19e6fe3d887bc5dcac7ca18d46199695463fdd                0.023   

      2017-01-01 00:30:00  2017-01-01 01:00:00  2017-01-01 01:30:00  \
0                     NaN                  NaN      

In [5]:
df.shape

(3248, 17521)

In [100]:
row=df.iloc[1]
begin,end=get_monthi(12)
print(row[begin:end])
print(row[begin:end].isnull().sum())
print(row[begin:end].values.any())
print(row[begin:end].mean())

2017-12-01 00:00:00      NaN
2017-12-01 00:30:00      NaN
2017-12-01 01:00:00      NaN
2017-12-01 01:30:00      NaN
2017-12-01 02:00:00      NaN
                       ...  
2017-12-31 21:00:00    0.868
2017-12-31 21:30:00    0.394
2017-12-31 22:00:00    0.073
2017-12-31 22:30:00    0.077
2017-12-31 23:00:00    0.049
Name: 1, Length: 1487, dtype: object
384
nan
0.3938087035358112


In [111]:
row=df.iloc[3247]
begin,end=get_monthi(11)
print(row[begin:end].shape)
print(end-begin)
print(row[begin:end].isnull().sum())

(1439,)
1439
48


In [112]:
# loop over all meter ids
for i in range(0,df.shape[0]):
    #load row
    meter=df.iloc[i]
    
    # loop over all months
    print("Meter i=%i has not NaN values in month:" %i)
    for m in range (1,13):
        ind_b,ind_e=get_monthi(m)               #get index of beginning and end of month
        month=meter[ind_b:ind_e]
        # Check if months has numeric values        
        n_NaN=month.isnull().sum() #Number of NaN's
        if n_NaN<(ind_e-ind_b) and m<12:
            print(m)


Meter i=0 has not NaN values in month:
Meter i=1 has not NaN values in month:
Meter i=2 has not NaN values in month:
Meter i=3 has not NaN values in month:
Meter i=4 has not NaN values in month:
Meter i=5 has not NaN values in month:
Meter i=6 has not NaN values in month:
Meter i=7 has not NaN values in month:
Meter i=8 has not NaN values in month:
Meter i=9 has not NaN values in month:
Meter i=10 has not NaN values in month:
Meter i=11 has not NaN values in month:
Meter i=12 has not NaN values in month:
Meter i=13 has not NaN values in month:
Meter i=14 has not NaN values in month:
Meter i=15 has not NaN values in month:
Meter i=16 has not NaN values in month:
Meter i=17 has not NaN values in month:
Meter i=18 has not NaN values in month:
Meter i=19 has not NaN values in month:
Meter i=20 has not NaN values in month:
Meter i=21 has not NaN values in month:
Meter i=22 has not NaN values in month:
Meter i=23 has not NaN values in month:
Meter i=24 has not NaN values in month:
Meter i=25

Meter i=204 has not NaN values in month:
Meter i=205 has not NaN values in month:
Meter i=206 has not NaN values in month:
Meter i=207 has not NaN values in month:
Meter i=208 has not NaN values in month:
Meter i=209 has not NaN values in month:
Meter i=210 has not NaN values in month:
Meter i=211 has not NaN values in month:
Meter i=212 has not NaN values in month:
Meter i=213 has not NaN values in month:
Meter i=214 has not NaN values in month:
Meter i=215 has not NaN values in month:
Meter i=216 has not NaN values in month:
Meter i=217 has not NaN values in month:
Meter i=218 has not NaN values in month:
Meter i=219 has not NaN values in month:
Meter i=220 has not NaN values in month:
Meter i=221 has not NaN values in month:
Meter i=222 has not NaN values in month:
Meter i=223 has not NaN values in month:
Meter i=224 has not NaN values in month:
Meter i=225 has not NaN values in month:
Meter i=226 has not NaN values in month:
Meter i=227 has not NaN values in month:
Meter i=228 has 

11
Meter i=398 has not NaN values in month:
11
Meter i=399 has not NaN values in month:
11
Meter i=400 has not NaN values in month:
11
Meter i=401 has not NaN values in month:
11
Meter i=402 has not NaN values in month:
11
Meter i=403 has not NaN values in month:
11
Meter i=404 has not NaN values in month:
11
Meter i=405 has not NaN values in month:
11
Meter i=406 has not NaN values in month:
11
Meter i=407 has not NaN values in month:
11
Meter i=408 has not NaN values in month:
11
Meter i=409 has not NaN values in month:
11
Meter i=410 has not NaN values in month:
11
Meter i=411 has not NaN values in month:
11
Meter i=412 has not NaN values in month:
11
Meter i=413 has not NaN values in month:
11
Meter i=414 has not NaN values in month:
11
Meter i=415 has not NaN values in month:
11
Meter i=416 has not NaN values in month:
11
Meter i=417 has not NaN values in month:
11
Meter i=418 has not NaN values in month:
11
Meter i=419 has not NaN values in month:
11
Meter i=420 has not NaN value

10
11
Meter i=584 has not NaN values in month:
10
11
Meter i=585 has not NaN values in month:
10
11
Meter i=586 has not NaN values in month:
10
11
Meter i=587 has not NaN values in month:
10
11
Meter i=588 has not NaN values in month:
10
11
Meter i=589 has not NaN values in month:
10
11
Meter i=590 has not NaN values in month:
10
11
Meter i=591 has not NaN values in month:
10
11
Meter i=592 has not NaN values in month:
10
11
Meter i=593 has not NaN values in month:
10
11
Meter i=594 has not NaN values in month:
10
11
Meter i=595 has not NaN values in month:
10
11
Meter i=596 has not NaN values in month:
10
11
Meter i=597 has not NaN values in month:
10
11
Meter i=598 has not NaN values in month:
10
11
Meter i=599 has not NaN values in month:
10
11
Meter i=600 has not NaN values in month:
10
11
Meter i=601 has not NaN values in month:
10
11
Meter i=602 has not NaN values in month:
10
11
Meter i=603 has not NaN values in month:
10
11
Meter i=604 has not NaN values in month:
10
11
Meter i

Meter i=761 has not NaN values in month:
10
11
Meter i=762 has not NaN values in month:
10
11
Meter i=763 has not NaN values in month:
10
11
Meter i=764 has not NaN values in month:
10
11
Meter i=765 has not NaN values in month:
10
11
Meter i=766 has not NaN values in month:
10
11
Meter i=767 has not NaN values in month:
10
11
Meter i=768 has not NaN values in month:
10
11
Meter i=769 has not NaN values in month:
10
11
Meter i=770 has not NaN values in month:
10
11
Meter i=771 has not NaN values in month:
10
11
Meter i=772 has not NaN values in month:
10
11
Meter i=773 has not NaN values in month:
10
11
Meter i=774 has not NaN values in month:
10
11
Meter i=775 has not NaN values in month:
10
11
Meter i=776 has not NaN values in month:
10
11
Meter i=777 has not NaN values in month:
10
11
Meter i=778 has not NaN values in month:
10
11
Meter i=779 has not NaN values in month:
10
11
Meter i=780 has not NaN values in month:
10
11
Meter i=781 has not NaN values in month:
10
11
Meter i=782 h

Meter i=939 has not NaN values in month:
9
10
11
Meter i=940 has not NaN values in month:
9
10
11
Meter i=941 has not NaN values in month:
9
10
11
Meter i=942 has not NaN values in month:
9
10
11
Meter i=943 has not NaN values in month:
9
10
11
Meter i=944 has not NaN values in month:
9
10
11
Meter i=945 has not NaN values in month:
9
10
11
Meter i=946 has not NaN values in month:
9
10
11
Meter i=947 has not NaN values in month:
9
10
11
Meter i=948 has not NaN values in month:
9
10
11
Meter i=949 has not NaN values in month:
9
10
11
Meter i=950 has not NaN values in month:
9
10
11
Meter i=951 has not NaN values in month:
9
10
11
Meter i=952 has not NaN values in month:
9
10
11
Meter i=953 has not NaN values in month:
9
10
11
Meter i=954 has not NaN values in month:
9
10
11
Meter i=955 has not NaN values in month:
9
10
11
Meter i=956 has not NaN values in month:
9
10
11
Meter i=957 has not NaN values in month:
9
10
11
Meter i=958 has not NaN values in month:
9
10
11
Meter i=959 has not 

8
9
10
11
Meter i=1110 has not NaN values in month:
8
9
10
11
Meter i=1111 has not NaN values in month:
8
9
10
11
Meter i=1112 has not NaN values in month:
8
9
10
11
Meter i=1113 has not NaN values in month:
8
9
10
11
Meter i=1114 has not NaN values in month:
8
9
10
11
Meter i=1115 has not NaN values in month:
8
9
10
11
Meter i=1116 has not NaN values in month:
8
9
10
11
Meter i=1117 has not NaN values in month:
8
9
10
11
Meter i=1118 has not NaN values in month:
8
9
10
11
Meter i=1119 has not NaN values in month:
8
9
10
11
Meter i=1120 has not NaN values in month:
8
9
10
11
Meter i=1121 has not NaN values in month:
8
9
10
11
Meter i=1122 has not NaN values in month:
8
9
10
11
Meter i=1123 has not NaN values in month:
8
9
10
11
Meter i=1124 has not NaN values in month:
8
9
10
11
Meter i=1125 has not NaN values in month:
8
9
10
11
Meter i=1126 has not NaN values in month:
8
9
10
11
Meter i=1127 has not NaN values in month:
8
9
10
11
Meter i=1128 has not NaN values in month:
8
9
10
11
Me

Meter i=1268 has not NaN values in month:
8
9
10
11
Meter i=1269 has not NaN values in month:
8
9
10
11
Meter i=1270 has not NaN values in month:
8
9
10
11
Meter i=1271 has not NaN values in month:
8
9
10
11
Meter i=1272 has not NaN values in month:
8
9
10
11
Meter i=1273 has not NaN values in month:
8
9
10
11
Meter i=1274 has not NaN values in month:
8
9
10
11
Meter i=1275 has not NaN values in month:
8
9
10
11
Meter i=1276 has not NaN values in month:
8
9
10
11
Meter i=1277 has not NaN values in month:
8
9
10
11
Meter i=1278 has not NaN values in month:
8
9
10
11
Meter i=1279 has not NaN values in month:
8
9
10
11
Meter i=1280 has not NaN values in month:
8
9
10
11
Meter i=1281 has not NaN values in month:
8
9
10
11
Meter i=1282 has not NaN values in month:
8
9
10
11
Meter i=1283 has not NaN values in month:
8
9
10
11
Meter i=1284 has not NaN values in month:
8
9
10
11
Meter i=1285 has not NaN values in month:
8
9
10
11
Meter i=1286 has not NaN values in month:
8
9
10
11
Meter i=1287

7
8
9
10
11
Meter i=1430 has not NaN values in month:
7
8
9
10
11
Meter i=1431 has not NaN values in month:
7
8
9
10
11
Meter i=1432 has not NaN values in month:
7
8
9
10
11
Meter i=1433 has not NaN values in month:
7
8
9
10
11
Meter i=1434 has not NaN values in month:
7
8
9
10
11
Meter i=1435 has not NaN values in month:
7
8
9
10
11
Meter i=1436 has not NaN values in month:
7
8
9
10
11
Meter i=1437 has not NaN values in month:
7
8
9
10
11
Meter i=1438 has not NaN values in month:
7
8
9
10
11
Meter i=1439 has not NaN values in month:
7
8
9
10
11
Meter i=1440 has not NaN values in month:
7
8
9
10
11
Meter i=1441 has not NaN values in month:
7
8
9
10
11
Meter i=1442 has not NaN values in month:
7
8
9
10
11
Meter i=1443 has not NaN values in month:
7
8
9
10
11
Meter i=1444 has not NaN values in month:
7
8
9
10
11
Meter i=1445 has not NaN values in month:
7
8
9
10
11
Meter i=1446 has not NaN values in month:
7
8
9
10
11
Meter i=1447 has not NaN values in month:
7
8
9
10
11
Meter i=1448 has

7
8
9
10
11
Meter i=1585 has not NaN values in month:
7
8
9
10
11
Meter i=1586 has not NaN values in month:
7
8
9
10
11
Meter i=1587 has not NaN values in month:
7
8
9
10
11
Meter i=1588 has not NaN values in month:
7
8
9
10
11
Meter i=1589 has not NaN values in month:
7
8
9
10
11
Meter i=1590 has not NaN values in month:
7
8
9
10
11
Meter i=1591 has not NaN values in month:
7
8
9
10
11
Meter i=1592 has not NaN values in month:
7
8
9
10
11
Meter i=1593 has not NaN values in month:
7
8
9
10
11
Meter i=1594 has not NaN values in month:
7
8
9
10
11
Meter i=1595 has not NaN values in month:
7
8
9
10
11
Meter i=1596 has not NaN values in month:
7
8
9
10
11
Meter i=1597 has not NaN values in month:
7
8
9
10
11
Meter i=1598 has not NaN values in month:
7
8
9
10
11
Meter i=1599 has not NaN values in month:
7
8
9
10
11
Meter i=1600 has not NaN values in month:
7
8
9
10
11
Meter i=1601 has not NaN values in month:
7
8
9
10
11
Meter i=1602 has not NaN values in month:
7
8
9
10
11
Meter i=1603 has

Meter i=1739 has not NaN values in month:
6
7
8
9
10
11
Meter i=1740 has not NaN values in month:
6
7
8
9
10
11
Meter i=1741 has not NaN values in month:
6
7
8
9
10
11
Meter i=1742 has not NaN values in month:
6
7
8
9
10
11
Meter i=1743 has not NaN values in month:
6
7
8
9
10
11
Meter i=1744 has not NaN values in month:
6
7
8
9
10
11
Meter i=1745 has not NaN values in month:
6
7
8
9
10
11
Meter i=1746 has not NaN values in month:
6
7
8
9
10
11
Meter i=1747 has not NaN values in month:
6
7
8
9
10
11
Meter i=1748 has not NaN values in month:
6
7
8
9
10
11
Meter i=1749 has not NaN values in month:
6
7
8
9
10
11
Meter i=1750 has not NaN values in month:
6
7
8
9
10
11
Meter i=1751 has not NaN values in month:
6
7
8
9
10
11
Meter i=1752 has not NaN values in month:
6
7
8
9
10
11
Meter i=1753 has not NaN values in month:
6
7
8
9
10
11
Meter i=1754 has not NaN values in month:
6
7
8
9
10
11
Meter i=1755 has not NaN values in month:
6
7
8
9
10
11
Meter i=1756 has not NaN values in month:
6
7
8


11
Meter i=1887 has not NaN values in month:
6
7
8
9
10
11
Meter i=1888 has not NaN values in month:
6
7
8
9
10
11
Meter i=1889 has not NaN values in month:
6
7
8
9
10
11
Meter i=1890 has not NaN values in month:
6
7
8
9
10
11
Meter i=1891 has not NaN values in month:
6
7
8
9
10
11
Meter i=1892 has not NaN values in month:
6
7
8
9
10
11
Meter i=1893 has not NaN values in month:
6
7
8
9
10
11
Meter i=1894 has not NaN values in month:
6
7
8
9
10
11
Meter i=1895 has not NaN values in month:
6
7
8
9
10
11
Meter i=1896 has not NaN values in month:
6
7
8
9
10
11
Meter i=1897 has not NaN values in month:
5
6
7
8
9
10
11
Meter i=1898 has not NaN values in month:
5
6
7
8
9
10
11
Meter i=1899 has not NaN values in month:
5
6
7
8
9
10
11
Meter i=1900 has not NaN values in month:
5
6
7
8
9
10
11
Meter i=1901 has not NaN values in month:
5
6
7
8
9
10
11
Meter i=1902 has not NaN values in month:
5
6
7
8
9
10
11
Meter i=1903 has not NaN values in month:
5
6
7
8
9
10
11
Meter i=1904 has not NaN values

Meter i=2036 has not NaN values in month:
5
6
7
8
9
10
11
Meter i=2037 has not NaN values in month:
5
6
7
8
9
10
11
Meter i=2038 has not NaN values in month:
5
6
7
8
9
10
11
Meter i=2039 has not NaN values in month:
5
6
7
8
9
10
11
Meter i=2040 has not NaN values in month:
5
6
7
8
9
10
11
Meter i=2041 has not NaN values in month:
5
6
7
8
9
10
11
Meter i=2042 has not NaN values in month:
5
6
7
8
9
10
11
Meter i=2043 has not NaN values in month:
5
6
7
8
9
10
11
Meter i=2044 has not NaN values in month:
5
6
7
8
9
10
11
Meter i=2045 has not NaN values in month:
5
6
7
8
9
10
11
Meter i=2046 has not NaN values in month:
5
6
7
8
9
10
11
Meter i=2047 has not NaN values in month:
5
6
7
8
9
10
11
Meter i=2048 has not NaN values in month:
5
6
7
8
9
10
11
Meter i=2049 has not NaN values in month:
5
6
7
8
9
10
11
Meter i=2050 has not NaN values in month:
5
6
7
8
9
10
11
Meter i=2051 has not NaN values in month:
5
6
7
8
9
10
11
Meter i=2052 has not NaN values in month:
5
6
7
8
9
10
11
Meter i=2053 h

4
5
6
7
8
9
10
11
Meter i=2184 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2185 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2186 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2187 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2188 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2189 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2190 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2191 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2192 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2193 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2194 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2195 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2196 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2197 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2198 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2199 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2200 has not N

Meter i=2327 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2328 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2329 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2330 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2331 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2332 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2333 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2334 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2335 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2336 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2337 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2338 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2339 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2340 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2341 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2342 has not NaN values in month:
4
5
6
7
8
9
10
11
Meter i=2343 has not NaN values in month

11
Meter i=2471 has not NaN values in month:
3
4
5
6
7
8
9
10
11
Meter i=2472 has not NaN values in month:
3
4
5
6
7
8
9
10
11
Meter i=2473 has not NaN values in month:
3
4
5
6
7
8
9
10
11
Meter i=2474 has not NaN values in month:
3
4
5
6
7
8
9
10
11
Meter i=2475 has not NaN values in month:
3
4
5
6
7
8
9
10
11
Meter i=2476 has not NaN values in month:
3
4
5
6
7
8
9
10
11
Meter i=2477 has not NaN values in month:
3
4
5
6
7
8
9
10
11
Meter i=2478 has not NaN values in month:
3
4
5
6
7
8
9
10
11
Meter i=2479 has not NaN values in month:
3
4
5
6
7
8
9
10
11
Meter i=2480 has not NaN values in month:
3
4
5
6
7
8
9
10
11
Meter i=2481 has not NaN values in month:
3
4
5
6
7
8
9
10
11
Meter i=2482 has not NaN values in month:
3
4
5
6
7
8
9
10
11
Meter i=2483 has not NaN values in month:
3
4
5
6
7
8
9
10
11
Meter i=2484 has not NaN values in month:
3
4
5
6
7
8
9
10
11
Meter i=2485 has not NaN values in month:
3
4
5
6
7
8
9
10
11
Meter i=2486 has not NaN values in month:
3
4
5
6
7
8
9
10
11
Meter

8
9
10
11
Meter i=2739 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2740 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2741 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2742 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2743 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2744 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2745 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2746 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2747 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2748 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2749 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2750 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2751 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2752 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2753 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2754 has not NaN value

2
3
4
5
6
7
8
9
10
11
Meter i=2874 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2875 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2876 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2877 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2878 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2879 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2880 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2881 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2882 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2883 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2884 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2885 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2886 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2887 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2888 has not NaN values in month:
2
3
4
5
6
7
8
9
10
11
Meter i=2889 has n

1
2
3
4
5
6
7
8
9
10
11
Meter i=3003 has not NaN values in month:
1
2
3
4
5
6
7
8
9
10
11
Meter i=3004 has not NaN values in month:
1
2
3
4
5
6
7
8
9
10
11
Meter i=3005 has not NaN values in month:
1
2
3
4
5
6
7
8
9
10
11
Meter i=3006 has not NaN values in month:
1
2
3
4
5
6
7
8
9
10
11
Meter i=3007 has not NaN values in month:
1
2
3
4
5
6
7
8
9
10
11
Meter i=3008 has not NaN values in month:
1
2
3
4
5
6
7
8
9
10
11
Meter i=3009 has not NaN values in month:
1
2
3
4
5
6
7
8
9
10
11
Meter i=3010 has not NaN values in month:
1
2
3
4
5
6
7
8
9
10
11
Meter i=3011 has not NaN values in month:
1
2
3
4
5
6
7
8
9
10
11
Meter i=3012 has not NaN values in month:
1
2
3
4
5
6
7
8
9
10
11
Meter i=3013 has not NaN values in month:
1
2
3
4
5
6
7
8
9
10
11
Meter i=3014 has not NaN values in month:
1
2
3
4
5
6
7
8
9
10
11
Meter i=3015 has not NaN values in month:
1
2
3
4
5
6
7
8
9
10
11
Meter i=3016 has not NaN values in month:
1
2
3
4
5
6
7
8
9
10
11
Meter i=3017 has not NaN values in month:
1
2
3
4
5


Meter i=3131 has not NaN values in month:
1
2
3
4
5
6
7
8
9
10
11
Meter i=3132 has not NaN values in month:
1
2
3
4
5
6
7
8
9
10
11
Meter i=3133 has not NaN values in month:
1
2
3
4
5
6
7
8
9
10
11
Meter i=3134 has not NaN values in month:
1
2
3
4
5
6
7
8
9
10
11
Meter i=3135 has not NaN values in month:
1
2
3
4
5
6
7
8
9
10
11
Meter i=3136 has not NaN values in month:
1
2
3
4
5
6
7
8
9
10
11
Meter i=3137 has not NaN values in month:
1
2
3
4
5
6
7
8
9
10
11
Meter i=3138 has not NaN values in month:
1
2
3
4
5
6
7
8
9
10
11
Meter i=3139 has not NaN values in month:
1
2
3
4
5
6
7
8
9
10
11
Meter i=3140 has not NaN values in month:
1
2
3
4
5
6
7
8
9
10
11
Meter i=3141 has not NaN values in month:
1
2
3
4
5
6
7
8
9
10
11
Meter i=3142 has not NaN values in month:
1
2
3
4
5
6
7
8
9
10
11
Meter i=3143 has not NaN values in month:
1
2
3
4
5
6
7
8
9
10
11
Meter i=3144 has not NaN values in month:
1
2
3
4
5
6
7
8
9
10
11
Meter i=3145 has not NaN values in month:
1
2
3
4
5
6
7
8
9
10
11
Meter i=31

In [93]:
def get_monthi(n):
    begin=48*31*(n-1)+1
    end=48*31*n
    if n==1:
        begin=1
    if n>1:
        end-=3*48
    if n>2:
        begin-=3*48
    if n>3:
        end-=48
    if n>4:
        begin-=48
    if n>5:
        end-=48
    if n>6:
        begin-=48
    if n>8:
        end-=48
    if n>9:
        begin-=48
    if n>10:
        end-=48
    if n>11:
        begin-=48
    return begin,end

Check performance naive drift

In [128]:
pli=False #print log info
NaN_t=1200 #NaN threshold (entire month has about 1400 datapoint)->we want atleast 1/7th of a month in this case
totSSE=0
nSSE=0


# loop over all meter ids
for i in range(0,df.shape[0]):
    #load row
    meter=df.iloc[i]
    fmf=False #first month found -> this variable is needed because naive drift needs 2 data points
    
    # loop over all months
    for m in range (1,13):
        ind_b,ind_e=get_monthi(m)               #get index of beginning and end of month
        month=meter[ind_b:ind_e]

        # Check if months has numeric values        
        n_NaN=month.isnull().sum() #Number of NaN's
        if n_NaN<NaN_t:
            mean_month=month.mean()
            
            #determine naive drift
            if fmf:
                ndrift=2*mean_month-last_month
                #evaluate prediction
                if m <12:
                    next_ind_b,next_ind_e=get_monthi(m+1) 
                    next_month=meter[next_ind_b:next_ind_e]
                    mnm=next_month.mean()         #mean next month
                    SSE=(mnm-ndrift)**2
                    totSSE+=SSE
                    nSSE+=1
                    if pli:
                        print("i=%i m=%i SSE=%f" % (i, m, SSE))
            else:
                fmf=True
            
            last_month=mean_month
              
        
        if pli:     
            print("Current month is %i" % m)
            print("Amount of NaN found %i" %n_NaN)


In [129]:
RMSE=np.sqrt(totSSE/nSSE)
print(RMSE)

0.06684300400401268


RMSE=0.06684300400401268
Average half hourly consumption of a month is between 0.16 and 0.30. Then our relative error is about 20 to 37.5 % 